File: zCompanies/zDRW/process.cpp
#include <iostream>
#include <fstream>
#include <vector>
#include <string>
#include <algorithm>

void processFiles(const std::string& basePath, const std::string& fillPath, const std::string& removePath, const std::string& outputPath) {
    // Read fill_file
    std::ifstream fillFile(fillPath);
    std::string fillLine;
    std::getline(fillFile, fillLine);
    fillFile.close();

    // Remove newline characters and trailing whitespace from fillLine
    fillLine.erase(std::remove(fillLine.begin(), fillLine.end(), '\r'), fillLine.end());
    fillLine.erase(std::remove(fillLine.begin(), fillLine.end(), '\n'), fillLine.end());
    fillLine.erase(std::remove_if(fillLine.begin(), fillLine.end(), ::isspace), fillLine.end());

    // Read remove_file
    std::ifstream removeFile(removePath);
    std::string removeLine;
    if (std::getline(removeFile, removeLine)) {
        // Remove newline characters and trailing whitespace from removeLine
        removeLine.erase(std::remove(removeLine.begin(), removeLine.end(), '\r'), removeLine.end());
        removeLine.erase(std::remove(removeLine.begin(), removeLine.end(), '\n'), removeLine.end());
        removeLine.erase(std::remove_if(removeLine.begin(), removeLine.end(), ::isspace), removeLine.end());
    }
    removeFile.close();

    // Process base_file
    std::ifstream baseFile(basePath);
    std::vector<std::string> baseLines;
    std::string line;
    while (std::getline(baseFile, line)) {
        baseLines.push_back(line);
    }
    baseFile.close();

    std::vector<std::string> modifiedLines;
    int emptyLineCount = 0;

    for (const std::string& origLine : baseLines) {
        std::string line = origLine;
        
        if (line.empty()) {
            emptyLineCount++;
            line = fillLine;
        } else if (!removeLine.empty()) {
            size_t pos = line.find(removeLine);
            while (pos != std::string::npos) {
                line.replace(pos, removeLine.length(), "");
                pos = line.find(removeLine, pos);
            }
        }
        
        modifiedLines.push_back(line);
    }

    // Write the modifications to output_file
    std::ofstream outputFile(outputPath);
    outputFile << "There are " << modifiedLines.size() << " lines in base file." << std::endl;
    outputFile << "There are " << emptyLineCount << " empty lines in base file." << std::endl;
    for (const std::string& modifiedLine : modifiedLines) {
        outputFile << modifiedLine << std::endl;
    }
    outputFile.close();
}

int main() {
    std::string basePath = "path_to_base_file.txt";
    std::string fillPath = "path_to_fill_file.txt";
    std::string removePath = "path_to_remove_file.txt";
    std::string outputPath = "path_to_output_file.txt";

    processFiles(basePath, fillPath, removePath, outputPath);

    return 0;
}

File: zCompanies/zDRW/process.py
def process_files(base_path, fill_path, remove_path, output_path):
    # Read fill_file
    with open(fill_path, 'r') as fill_file:
        fill_line = fill_file.readline().rstrip()

    # Remove newline characters and trailing whitespace from fill_line
    fill_line = fill_line.replace('\r', '').replace('\n', '').strip()

    # Read remove_file
    with open(remove_path, 'r') as remove_file:
        remove_line = remove_file.readline().rstrip() if remove_file else ""

    # Remove newline characters and trailing whitespace from remove_line
    remove_line = remove_line.replace('\r', '').replace('\n', '').strip()

    # Process base_file
    with open(base_path, 'r') as base_file:
        base_lines = base_file.readlines()

    modified_lines = []
    empty_line_count = 0

    for orig_line in base_lines:
        line = orig_line.rstrip()

        if not line:
            empty_line_count += 1
            line = fill_line
        elif remove_line:
            line = line.replace(remove_line, '')

        modified_lines.append(line)

    # Write the modifications to output_file
    with open(output_path, 'w') as output_file:
        output_file.write(f"There are {len(modified_lines)} lines in base file.\n")
        output_file.write(f"There are {empty_line_count} empty lines in base file.\n")
        for modified_line in modified_lines:
            output_file.write(f"{modified_line}\n")

if __name__ == "__main__":
    base_path = "path_to_base_file.txt"
    fill_path = "path_to_fill_file.txt"
    remove_path = "path_to_remove_file.txt"
    output_path = "path_to_output_file.txt"

    process_files(base_path, fill_path, remove_path, output_path)